Submitted By: Andrew Bradford <andrew@bradfordembedded.com>
Date: 2013-10-16
Initial Package Version: 1.21.1
Origin: Buildroot git and BusyBox git
Upstream Status: ifconfig.c and partially iplink.c yes, libbb.h no
Description: Patches to BusyBox to build for musl-libc.

diff -Naur busybox-1.21.1-orig/include/libbb.h busybox-1.21.1/include/libbb.h
--- busybox-1.21.1-orig/include/libbb.h	2013-06-29 10:58:06.000000000 -0400
+++ busybox-1.21.1/include/libbb.h	2013-10-16 10:38:04.222276525 -0400
@@ -37,7 +37,7 @@
 #include <libgen.h> /* dirname,basename */
 #undef basename
 #define basename dont_use_basename
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/ioctl.h>
 #include <sys/mman.h>
 #include <sys/socket.h>
diff -Naur busybox-1.21.1-orig/networking/ifconfig.c busybox-1.21.1/networking/ifconfig.c
--- busybox-1.21.1-orig/networking/ifconfig.c	2013-05-11 19:30:43.000000000 -0400
+++ busybox-1.21.1/networking/ifconfig.c	2013-10-16 10:37:51.628945111 -0400
@@ -56,7 +56,7 @@
 #endif
 
 #if ENABLE_FEATURE_IFCONFIG_SLIP
-# include <net/if_slip.h>
+# include <linux/if_slip.h>
 #endif
 
 /* I don't know if this is needed for busybox or not.  Anyone? */
diff -Naur busybox-1.21.1-orig/networking/libiproute/iplink.c busybox-1.21.1/networking/libiproute/iplink.c
--- busybox-1.21.1-orig/networking/libiproute/iplink.c	2013-05-11 19:30:43.000000000 -0400
+++ busybox-1.21.1/networking/libiproute/iplink.c	2013-10-16 10:59:59.925471509 -0400
@@ -5,8 +5,7 @@
  * Licensed under GPLv2 or later, see file LICENSE in this source tree.
  */
 #include <net/if.h>
-#include <net/if_packet.h>
-#include <netpacket/packet.h>
+#include <linux/if_packet.h>
 #include <netinet/if_ether.h>
 
 #include "ip_common.h"  /* #include "libbb.h" is inside */
